### ISSUE TITLE
Support for psql ON CONFLICT

### ISSUE THREAD

**Issue by mberlanda** [jamis/bulk_insert#17]
Would it be possible to extend the `ignore: true` option for PostgreSQL adapter as well?
When I added it to my Rails application, I got the following error:

```
ActiveRecord::StatementInvalid: PG::SyntaxError: ERROR:  syntax error at or near "IGNORE"
LINE 1: INSERT IGNORE INTO "table_name" ("date","i...
```

As a suggestion/benchmark, you can find below the way it was implemented in [`activerecord-import gem`](https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/adapters/postgresql_adapter.rb):

```rb
# postgresql_adapter.rb
# ...
  def post_sql_statements( table_name, options ) # :nodoc:
    sql = []

    if supports_on_duplicate_key_update?
      # Options :recursive and :on_duplicate_key_ignore are mutually exclusive
      if (options[:ignore] || options[:on_duplicate_key_ignore]) && !options[:on_duplicate_key_update] && !options[:recursive]
        sql << sql_for_on_duplicate_key_ignore( table_name, options[:on_duplicate_key_ignore] )
      end
    elsif options[:on_duplicate_key_ignore] && !options[:on_duplicate_key_update]
      logger.warn "Ignoring on_duplicate_key_ignore because it is not supported by the database."
    end

    sql += super(table_name, options)

    unless options[:no_returning] || options[:primary_key].blank?
      primary_key = Array(options[:primary_key])
      sql << " RETURNING \"#{primary_key.join('", "')}\""
    end

    sql
  end
  def sql_for_on_duplicate_key_ignore( table_name, *args ) # :nodoc:
    arg = args.first
    conflict_target = sql_for_conflict_target( arg ) if arg.is_a?( Hash )
    " ON CONFLICT #{conflict_target}DO NOTHING"
  end
```

I haven't found any CONTRIBUTING guideline in your project. Thank you

**Comment by jamis** [comment]
That's a great idea, and if you could implement that, I think it would make a great addition to BulkInsert.

As far as contributing goes, if you fork the project and make your changes (with tests) in your own fork of the repository, you can then submit your change(s) as a pull request, we could discuss it in more concrete terms at that point.

Thanks!

**Comment by mberlanda** [comment]
Hi jamis,  thank you for getting back to me. I forked the project and I wanted to make sure a couple of things before to start working on my fork:

- is it fine if my commit will be complaint with Rails 4 and 5 only? I saw in the [apidoc](http://apidock.com/rails/ActiveRecord/Base/connection) that `ActiveRecord::Base.connection` changed after Rails 3 and I assume that you are already referring to the new one. I should make the `ignore:` parameter depending on the database connection
- which gem are you using for tests? I usually prefer `rspec` so it would be useful a link to your test suite
- would you mind if I add `rvm` files to the project?  In case, which ruby version would you prefer to use for this project?
- would you agree to add `overcommit` hook with rubocop?

That's almost all I need in terms of contributing guidelines :)

**Comment by jamis** [comment]
@mberlanda:

* Yes, it's fine to support only Rails 4/5.
* The test suite uses just ActiveSupport::TestCase (which itself is based on Minitest, I believe?). The tests are all in the `test` directory.
* I don't know that it is necessary to add `rvm` files. The gem should be compliant with any Ruby version 2+.
* I have a real love/hate relationship with rubocop. I'm learning to appreciate many things about it, but I don't have time just now to define a consistent coding style for this project. If you'd like to propose one that is compatible (or mostly compatible) with the project's current style, we could use that as a starting point.

**Comment by mberlanda** [comment]
Resolved in pull request #19
